### PR TITLE
Feature/fab button

### PR DESCRIPTION
**Pull request by SolMendiola**
# :star: Feature

---

#### :pencil2: Description:

Added fab button to UI

---

#### iOS
Simulator Screen Shot - iPhone 13 - 2022-11-04 at 17 13 58
Simulator Screen Shot - iPhone 13 - 2022-11-04 at 17 13 48

---

### CODE REVIEW

**Review by f7deleon — COMMENTED** [review]
I think that it would be best to update the branch first so we can see what changes were done specifically to this feature.

File: lib/ui/tips/tips_screen.dart [hunk 1]
```diff
@@ -1,9 +1,12 @@
...
 import 'package:flutter_bloc/flutter_bloc.dart';
-import 'package:flutter_template/ui/extensions/context_extensions.dart';
 import 'package:flutter_template/ui/section/error_handler/error_handler_cubit.dart';
-import 'package:flutter_template/ui/theme/app_theme.dart';
 import 'package:flutter_template/ui/tips/tips_cubit.dart';
+import 'package:flutter_template/ui/common/custom_scaffold_fab.dart';
+import 'package:flutter_template/ui/common/fab.dart';
+
+import '../common/custom_scaffold_fab.dart';
+import '../common/fab.dart';
```
Comment (f7deleon) [inline lib/ui/tips/tips_screen.dart:9]: check this imports

File: lib/ui/tips/tips_cubit.dart [hunk 2]
```diff
@@ -23,6 +23,18 @@ class TipCubit extends Cubit<TipsBaseState> {
     unawaited(getTips());
   }
 
+  void setCurrentPage(int index) {
+    emit(state.copyWith(currentPage: index));
+  }
+
+  void changeFavouriteButton(int index) {
+    // state.tips[index].favourite
```
Comment (f7deleon) [inline lib/ui/tips/tips_cubit.dart:31]: check this

File: lib/ui/home/home_screen.dart [hunk 3]
```diff
@@ -35,13 +44,19 @@ class _HomeScreenState extends State<HomeScreen> {
...
-    return FloatingActionButton(
-      foregroundColor: context.theme.colors.primary,
-      backgroundColor: context.theme.colors.primary.shade100,
-      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16.r)),
-      onPressed: () => _scaffoldKey.currentState!.openDrawer(),
-      child: Icon(Icons.arrow_forward_ios),
+  Widget buildFab(BuildContext context) {
+    return Visibility(
+      visible: !hideFab,
+      child: FloatingActionButton(
```
Comment (f7deleon) [inline lib/ui/home/home_screen.dart:50]: what's the difference between this and the FAB class used on this [line](https://github.com/xmartlabs/ftnt/pull/22/files#diff-633666d311cec50fb8b6b95f713e527773a4f0cc147cbf7f7fc7b5359ab78a5aR27)?

File: lib/ui/tips/tips_cubit.dart [hunk 4]
```diff
@@ -23,6 +23,18 @@ class TipCubit extends Cubit<TipsBaseState> {
     unawaited(getTips());
   }
 
+  void setCurrentPage(int index) {
```
Comment (f7deleon) [inline lib/ui/tips/tips_cubit.dart:26]: change it to `=>
`

File: lib/ui/common/custom_scaffold_fab.dart [hunk 5]
```diff
@@ -0,0 +1,36 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_template/ui/extensions/context_extensions.dart';
+import 'package:flutter_template/ui/theme/app_theme.dart';
+import 'package:flutter_template/ui/common/fab.dart';
+
+class CustomScaffoldFab extends StatelessWidget {
```
Comment (mirland) [inline lib/ui/common/custom_scaffold_fab.dart:6]: MainScaffoldWithFab? I'm not sure, but I don't like the current name

File: lib/ui/home/home_screen.dart [hunk 6]
```diff
@@ -1,24 +1,34 @@
...
 }
 
 class _HomeScreenState extends State<HomeScreen> {
   final _scaffoldKey = GlobalKey<ScaffoldState>();
+  var hideFab = false;
+
+  @override
+  void initState() {
+    super.initState();
+  }
```
Comment (mirland) [inline lib/ui/home/home_screen.dart:21]: do you need it?

File: lib/ui/home/home_screen.dart [hunk 7]
```diff
@@ -1,24 +1,34 @@
...
+import 'package:flutter_template/ui/common/fab.dart';
+
 class HomeScreen extends StatefulWidget {
   @override
   State<HomeScreen> createState() => _HomeScreenState();
 }
 
 class _HomeScreenState extends State<HomeScreen> {
   final _scaffoldKey = GlobalKey<ScaffoldState>();
+  var hideFab = false;
```
Comment (mirland) [inline lib/ui/home/home_screen.dart:16]: private

File: lib/ui/home/home_screen.dart [hunk 8]
```diff
@@ -35,13 +45,14 @@ class _HomeScreenState extends State<HomeScreen> {
...
 
-  FloatingActionButton buildFab(BuildContext context) {
-    return FloatingActionButton(
-      foregroundColor: context.theme.colors.primary,
-      backgroundColor: context.theme.colors.primary.shade100,
-      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(16.r)),
-      onPressed: () => _scaffoldKey.currentState!.openDrawer(),
-      child: Icon(Icons.arrow_forward_ios),
+  Widget buildFab(BuildContext context) {
+    return Visibility(
```
Comment (mirland) [inline lib/ui/home/home_screen.dart:49]: lamda

File: lib/ui/tips/tips_cubit.dart [hunk 9]
```diff
@@ -23,6 +23,15 @@ class TipCubit extends Cubit<TipsBaseState> {
...
   }
 
+  void setCurrentPage(int index) => emit(state.copyWith(currentPage: index));
+
+  void changeFavouriteButton(int index) {
+    final newList = [...state.tips];
+    var tip = newList[index];
+    newList[index] = tip.copyWith(favourite: !tip.favourite);
+    emit(state.copyWith(tips: newList));
+  }
```
Comment (mirland) [inline lib/ui/tips/tips_cubit.dart:33]: this should be done in the repository later, maybe a todo?

File: lib/ui/videos/videos.dart [hunk 10]
```diff
@@ -12,6 +14,11 @@ class VideosScreen extends StatelessWidget {
 class _VideosContentScreen extends StatelessWidget {
   @override
   Widget build(BuildContext context) {
-    return Center(child: Text('VIDEOS'));
+    return Center(
+      child: Text(
+        'VIDEOS',
+        style: TextStyle(color: context.theme.colors.onPrimary),
```
Comment (mirland) [inline lib/ui/videos/videos.dart:20]: theme style?

File: lib/ui/common/custom_scaffold_fab.dart [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+    required this.state,
+    this.iconSelected,
+    required this.iconNotSelected,
+    this.action,
+    this.child,
+  }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
```
Comment (mirland) [inline lib/ui/common/custom_scaffold_fab.dart:24]: lamda

**Review by mirland — APPROVED** [review]

**Pull request merged**